eva_cttv_pipeline/clinvar_xml_utils.py

```diff
@@ -435,3 +438,64 @@ def sequence_location_helper(self, attr):
             # TODO: https://github.com/EBIvariation/eva-opentargets/issues/172
             return None
         return sequence_locations[0].attrib.get(attr)
+
+    def get_variant_name_or_hgvs(self):
+        if self.preferred_or_other_name:
+            return self.preferred_or_other_name
+        if self.toplevel_refseq_hgvs:
+            return self.toplevel_refseq_hgvs
+        return None
+
+    def _parse_identifier(self):
+        if self._parsed_identifier is None:
+            self._parsed_identifier = clinvar_identifier_parsing.parse_variant_identifier(self.get_variant_name_or_hgvs())
+        return self._parsed_identifier
```

Comment:
For these kind of construct I like using [cached-property](https://docs.python.org/3/library/functools.html#functools.cached_property). I think we use python 3.8 so it should be available from the standard library but otherwise we just need to add [this](https://pypi.org/project/cached-property/) to the requirement.txt

Reply:
Thank you! I have extracted the properties to a new class but I have use `@cached_property` on the property in this class as you suggested.